src/library/ziputils.lua

```diff
@@ -0,0 +1,225 @@
+You would normally call this on the temporary directory name from `calc_temp_output_path`.
+But it works on any directory.
+
+@ path_to_remove (string) platform-encoded path of directory to remove.
+: (string) platform-encoded command string to execute.
+]]
+function ziputils.calc_rmdir_command(path_to_remove)
+    return (finenv.UI():IsOnMac() and "rm -r " or "cmd /c rmdir /s /q ") .. path_to_remove
+end
+
+--[[
+% calc_delete_file_command
+
+Returns the platform-dependent command to delete a file. It can be passed
+to `luaosutils.process.execute`.
+
+**WARNING** The command, if executed, permanently deletes the file.
+You would normally call this on the temporary directory name from `calc_temp_output_path`.
+But it works on any directory.
+
+@ path_to_remove (string) platform-encoded path of directory to remove.
+: (string) platform-encoded command string to execute.
+]]
+function ziputils.calc_delete_file_command(path_to_remove)
+    return (finenv.UI():IsOnMac() and "rm " or "cmd /c del ") .. path_to_remove
+end
+
+
+--[[
+% calc_temp_output_path
+
+Returns a path that can be used as a temporary target for unzipping. The caller may create it
+either as a file or a directory, because it is guaranteed not to exist when it is returned and it does
+not have a terminating path delimiter. Also returns a platform-dependent unzip command that can be
+passed to `luaosutils.process.execute` to unzip the input archive into the temporary name as a directory.
+The command may not be compatible with `os.execute`.
+
+This function requires `luaosutils`.
+
+@ [archive_path] (string) platform-encoded filepath to the zip archive that is included in the zip command.
+: (string) platform-encoded temporary path generated by the system.
+: (string) platform-encoded unzip command that can be used to unzip a multifile archived directory structure into the temporary path.
+]]
+function ziputils.calc_temp_output_path(archive_path)
+    if not_supported_message then
+        error(not_supported_message, 2)
+    end
+
+    archive_path = archive_path or ""
+
+    local process = require("luaosutils").process
+
+    local output_dir = os.tmpname()
+    local rmcommand = ziputils.calc_delete_file_command(output_dir)
+    process.execute(rmcommand)
+
+    local zipcommand
+    if finenv.UI():IsOnMac() then
+        zipcommand = "unzip \"" .. archive_path .. "\" -d " .. output_dir
+    else
+        zipcommand = "cmd /c 7z x -o" .. output_dir .. " \"" .. archive_path .. "\""
+    end
+    return output_dir, zipcommand
+end
+
+--[[
+% calc_gunzip_command
+
+Returns the platform-dependent command to gunzip a file. It can be passed
+to `luaosutils.process.execute`, which will then return the text directly.
+
+
+@ archive_path (string) platform-encoded path of source gzip archive.
+: (string) platform-encoded command string to execute.
+]]
+function ziputils.calc_gunzip_command(archive_path)
```

Comment:
@rpatters1 You might consider using native Powershell commands instead of requiring 7z. Powershell can already deal directly with zip files using [Expand-Archive](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/expand-archive?view=powershell-5.1). It takes a [few lines of code](https://securitytidbits.wordpress.com/2017/04/14/powershell-and-gzip-compression/comment-page-1/) to deal with gzip.